android/src/app/organicmaps/SplashActivity.java

```diff
@@ -138,13 +151,9 @@ private void showFatalErrorDialog(@StringRes int titleId, @StringRes int message
         .show();
   }
 
-  @Override
-  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
-                                         @NonNull int[] grantResults)
+  @androidx.annotation.UiThread
+  private void onRequestPermissionsResult()
   {
-    if (requestCode != REQUEST_PERMISSIONS)
-      throw new AssertionError("Unexpected requestCode");
-    super.onRequestPermissionsResult(requestCode, permissions, grantResults);
     Config.setLocationRequested();
```

Comment:
Can this line of code be called directly instead of calling onRequestPermissionsResult() ?